### PR TITLE
Filter notes

### PR DESCRIPTION
**Pull request by Jay-Gadhiya**
Test link: https://cheer-notes-bqi7eh9as-jay-gadhiya.vercel.app/

functionality adds:  filter notes by priority and date. you can see notes by its tags category in the tags page.

### CODE REVIEW

File: src/Utilities-Functions/editNote.jsx [hunk 1]
```diff
@@ -9,11 +9,14 @@ const editNote = async (note, authState, noteDispatch, setUserNote) => {
             noteDispatch({type : "EDIT_NOTE", payload : res.data.notes });
             setUserNote(pre => ({ ...pre, title : "", content : "", color : "",flag : false, tags : [] }));
         }
-        
+
     } catch (error) {
         return error;
     
     }
+
```
Comment (aamir74) [inline src/Utilities-Functions/editNote.jsx:17]: Can remove extra blank lines.

**Review by aamir74 — APPROVED** [review]
Code is well written.
Filters are working perfectly.

File: src/components/Aside/Aside.jsx [hunk 2]
```diff
@@ -53,11 +83,71 @@ const Aside = () => {
...
+                        <label htmlFor="high">High</label>
+                    </div>
+                    <div className="priority-input">
+                        <input 
+                        type="radio" 
+                        name="priority"
+                        checked={filterNote.priority === "Medium"}
+                        onChange={() => setMediumPriority()}
+                         />
+                        <label htmlFor="medium">Medium</label>
```
Comment (AnkurChunekar) [inline src/components/Aside/Aside.jsx:110]: Include `id` attribute for all the inputs

File: src/Reducers/notesActions-reducer.jsx [hunk 3]
```diff
@@ -30,7 +30,27 @@ const notesReducer = (state, action) => {
...
+
+        case "PRIORITY_LOW":
+            return {...state, notes : state.allNotesData.filter(item => item.priority === action.payload)};
+
+        case "FILTER_NEWEST_FIRST":
+            console.log("new")
+            return {...state, notes : state.notes.sort((a, b) => Number(b.time) - Number(a.time))};
+
+        case "FILTER_OLDEST_FIRST":
+            console.log("old")
```
Comment (AnkurChunekar) [inline src/Reducers/notesActions-reducer.jsx:49]: Remove Console logs

File: src/components/Card/NotesCard.jsx [hunk 4]
```diff
@@ -16,20 +16,25 @@ import { deleteChip } from "../../Utilities-Functions/deleteChip";
...
     const { authState } = useAuth();
-    const { noteDispatch, setUserNote, userNote, color, setColor } = useNote();
+    const { noteDispatch, setUserNote, setFilterNote, filterNote } = useNote();
     const [colorPalate, setColorPalate] = useState(false);
     const [tagBox, setTagBox] = useState(false);
     let itemTags = [...note.tags];
     let tempTag = "";
 
     const changeInputs = (note) => {
         console.log("note", note);
```
Comment (AnkurChunekar) [inline src/components/Card/NotesCard.jsx:26]: Remove it from here too

File: src/components/Aside/Aside.css [hunk 5]
```diff
@@ -53,3 +53,44 @@
     font-size: 2rem;
     margin-left: 1rem;
 }
+
+
+/* filter */
+
+.filter-container {
+    /* border: 1px solid black; */
```
Comment (AnkurChunekar) [inline src/components/Aside/Aside.css:61]: Remove Unnecessary comments

File: src/Utilities-Functions/deleteNote.jsx [hunk 6]
```diff
@@ -9,6 +9,8 @@ const deleteNote = async (item, authState, noteDispatch, setUserNote) => {
             noteDispatch({type : "DELETE_NOTE", payload : res.data.notes });
             noteDispatch({type : "TRASH_NOTE", payload : item});
             setUserNote(pre => ({ ...pre, title : "", content : "", flag : false }));
+            setFilterNote(pre => ({...pre, priority : "", sortByDate : ""}));
```
Comment (AnkurChunekar) [inline src/Utilities-Functions/deleteNote.jsx:12]: Namings can be improved: you can use `prevState` instead of `pre`

**Review by jeetbhalani15 — APPROVED** [review]
-  Nicely used Reducer & context for filtering its works fine.
-  Liked Filter component UI looks clean.
- Rest all looks good ~ Nice Work bro !!

File: src/Reducers/notesActions-reducer.jsx [hunk 7]
```diff
@@ -1,21 +1,21 @@
...
 
         case "EDIT_NOTE":
-            return {...state, notes : action.payload };
+            return {...state, notes : action.payload, allNotesData : action.payload };
 
         case "ARCHIVE_NOTE":
-            return {...state, notes : action.payload.notes, archives : action.payload.archives };
+            return {...state, notes : action.payload.notes, archives : action.payload.archives, allNotesData : action.payload.notes };
 
         case "RESTOR_ARCHIVE_NOTE":
```
Comment (AnkurChunekar) [inline src/Reducers/notesActions-reducer.jsx:17]: if it is supposed to be Restore, then you can correct the spelling `RESTORE_ARCHIVE_NOTE, can avoid confusion.

File: src/Utilities-Functions/addNote.jsx [hunk 8]
```diff
@@ -7,7 +7,7 @@ const addNote = async (note, authState, noteDispatch, setUserNote) => {
         const res = await axios.post("/api/notes", { note }, { headers : { authorization: authState.token } }); 
         if(res.status === 201){
             setUserNote(pre => ({ ...pre, title : "", content : "", color : "" }));
-            noteDispatch({type : "ADD_NOTE", payload : res.data.notes });
+            noteDispatch({type : "ADD_NOTE", payload : res.data.notes }); 
         }
```
Comment (AnkurChunekar) [inline src/Utilities-Functions/addNote.jsx:13]: Indentation should be consistent.

**Review by AnkurChunekar — APPROVED** [review]
Other than the suggestions above, code looks good to merge.

**Pull request merged**